src/util/sandbox.cpp

```diff
@@ -369,80 +369,119 @@ QString Sandbox::migrateOldSettings() {
+    QString sandboxedParentPath = homePath +
             QLatin1String(
-                    "/Library/Containers/org.mixxx.mixxx/Data/Library/Application Support/Mixxx");
-    QDir sandboxedSettings(sandboxedPath);
+                    "/Library/Containers/org.mixxx.mixxx/Data/Library/"
+                    "Application Support");
+    QString sandboxedPath = sandboxedParentPath + QLatin1String("/Mixxx");
+    QDir sandboxedDir(sandboxedPath);
+
+    QString legacySettingsPath = homePath + QLatin1String("/Library/Application Support/Mixxx");
+    // The user has no settings from Mixxx < 2.3.0, so there is no migration to do.
+    if (!QDir(legacySettingsPath).exists()) {
+        return sandboxedPath;
+    }
 
-    if (sandboxedSettings.exists() && !sandboxedSettings.isEmpty()) {
+    // The user already has settings in the sandboxed path, so there is no migration to do.
+    if (sandboxedDir.exists() && !sandboxedDir.isEmpty()) {
         return sandboxedPath;
     }
 
-    // Because Mixxx cannot test if the old path exists before getting permission to access it
-    // outside the sandbox, unfortunately it is necessary to annoy the user with this popup
-    // even if they are installing Mixxx >= 2.3.0 without having installed an old version of Mixxx.
+    // Sandbox::askForAccess cannot be used here because it depends on settings being
+    // initialized. There is no need to store the bookmark anyway because this is a
+    // one time process.
     QString title = QObject::tr("Upgrading old Mixxx settings");
-    QString oldPath = homePath + QLatin1String("/Library/Application Support/Mixxx");
     QMessageBox::information(nullptr,
             title,
             QObject::tr(
-                    "Due to Mac sandboxing, Mixxx needs your permission to "
-                    "access your music library "
-                    "and settings from Mixxx versions before 2.3.0. After "
-                    "clicking OK, you will see a file picker. "
-                    "To give Mixxx permission, press the Ok button in the file "
-                    "picker."
+                    "Due to macOS sandboxing, Mixxx needs your permission "
+                    "to access your music library and settings from Mixxx "
+                    "versions before 2.3.0. After clicking OK, you will see a "
+                    "file selection dialog. "
                     "\n\n"
-                    "If you do not want to grant Mixxx access click Cancel on "
-                    "the file picker."));
+                    "To allow Mixxx to use your old library and settings, "
+                    "click the Open button in the file selection dialog. "
+                    "Mixxx will then move your old settings into the sandbox. "
+                    "This only needs to be done once."
+                    "\n\n"
+                    "If you do not want to grant Mixxx access, click Cancel "
+                    "on the file picker."));
```

Comment:
What happens in that case? Will mixxx start with a fresh database and config> Should we add that to the message box?

Reply:
Yeah, I'll make that more explicit.